doc/design/block_execution.md

```diff
@@ -0,0 +1,113 @@
+
+Let's consider about this question, What configuration should Paddle get from uses to execute a block in various hardward environment.
+
+  
+- Data Parallelism
+  - stand-alone machine: Users have to set several device ids, and each device will execute the same block.
+  - cluster: Users have to set node ids and device ids, and each device will execute the same block.
+  
+- Model Parallelism
+  - Users have to set a concrete device id for each operator in a topology. Paddle will first execute the operator strictly following users' configuration, and will do some another parallel optimization automatically.
+
+ 
+
+
+## Solution
+  
+Actually, we can have both data parallelism and model parallelism. In order to support the mixing paralelism, we have to propose these two strategies:
+
+
+- A simple device id setting rules
+  - Firsly, users have to set one or several default device id for a topology. These devices are the same and usually finish the most computation of a topology. We mainly consider data parallelism in this step.
+  - Secondly, users need to set device id for each operators in a topology. If users set nothing, the operator will just use default device id. When devices have to be switched, users have to add a copy operator manually.
+  - We travel the topology and set concrete device id for each operator.
+
+
+Here is a pre-sudo code about the device setting rules.
+
+
+```
+import paddle as pd
+
+def fpga_infer(fpga_id):
+    with pd.device(fpga_id):
+        x = pd.v2.variable(data_type='float32', dims=[32, 784], device='cpu')
+        w = pd.v2.variable(data_type='float32', dims=[784, 100], device='cpu')
+        x_fpga = pd.v2.copy_op(x)
+        w_fpga = pd.v2.copy_op(w)
+        h = pd.v2.mul_op(x_fpga, w_fgpa)
+        h_gpu = pd.v2.copy(h, device='gpu:0')
+        out = pd.v2.relu_op(h_gpu, device='gpu:0')
+        out_cpu = pd.v2.copy_op(out, device='cpu')
+    return out_cpu
+
+pd.run(fpga_infer, devices=['fpga:0', 'fpga:1'])    
+    
+```
+
+- A data dependency analysis engine
+  - We need to analysis data dependency of each operator in the topology. When the dependent variables of a operator is ready, the operator can be executed in corresponding device.
+
+  
+
+The concept `Block` contains full information for executing a neural network topology. And we will have another concept `Executor` which have a runtime data depenency analysis engine to execute operators in `Block` in certain hardware with high performance.
+  
+  
+## Conclusion
+
+We will split our implementation of `Block` into following steps:
```

Comment:
This section seems like a plan, and I think we should not put plan to a design doc, because when we implement part of the design we have to change this design doc. 

Design docs should be more "static", put plans to issue or projects may be better.

Can merge this sections to features we are going to implement.

Reply:
Project is more appropriate.